src/render_tasks/d3d12_deferred_composition.cpp

```diff
@@ -218,12 +218,18 @@ namespace wr
 
 				//GetSkybox
 				auto skybox = scene_graph.GetCurrentSkybox();
-				if (skybox != nullptr)
+				if (skybox)
 				{
 					//data.out_skybox = static_cast<wr::d3d12::TextureResource*>(pred_data.in_radiance.m_pool->GetTexture(pred_data.in_radiance.m_id));
 					data.out_skybox = static_cast<wr::d3d12::TextureResource*>(skybox->m_skybox->m_pool->GetTextureResource(skybox->m_skybox.value()));
 					d3d12::CreateSRVFromTexture(data.out_skybox);
 				}
+				else
```

Comment:
This is turning into really shitty duplicate code. We should probably make a function that returns the texture for us instead.

Reply:
Reworked everythig as discussed privately